manifests-gen/providers.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+	configclient "sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/repository"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/yamlprocessor"
+	utilyaml "sigs.k8s.io/cluster-api/util/yaml"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	powerVSProvider         = "powervs"
+	ibmCloudProvider        = "ibmcloud"
+	metadataFilePath        = "./metadata.yaml"
+	kustomizeComponentsPath = "./config/default"
+)
+
+type provider struct {
+	Name       string                    `json:"name"`
+	PType      clusterctlv1.ProviderType `json:"type"`
+	Version    string                    `json:"version"`
+	components repository.Components
+	metadata   []byte
+}
+
+// loadComponents loads components from the given provider.
+func (p *provider) loadComponents() error {
+	// Create new clusterctl config client
+	configClient, err := configclient.New("")
+	if err != nil {
+		return err
```

Comment:
Should wrap errors with additional context

Reply:
I didn't write any of this code, this is all code motion from https://github.com/openshift/cluster-capi-operator/blob/main/hack/assets/

Reply:
Same as https://github.com/openshift/cluster-capi-operator/pull/121#discussion_r1405973164

Reply:
Can add the wrapping 👍